presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java

```diff
@@ -146,7 +153,11 @@ public StoragePartitionLoader(
                 .map(p -> p.getColumns().size())
                 .orElse(table.getDataColumns().size());
         List<HivePartitionKey> partitionKeys = getPartitionKeys(table, partition.getPartition());
-        Path path = new Path(getPartitionLocation(table, partition.getPartition()));
+        String location = getPartitionLocation(table, partition.getPartition());
+        if (location.isEmpty() && table.getTableType().equals(TEMPORARY_TABLE) && !createEmptyBucketFilesForTemporaryTable) {
```

Comment:
Could you please elaborate on the case when it could be empty? Does it happen when the table is empty? If that's the case why does the partition location is expected to be an empty string vs a real path to an empty directory?

Reply:
Yes, it happens when a bucketed table is empty.  If we don't create any empty file for this table, empty directory will not be created (directory was created when we create a file). Should we create an empty directory for this scenario?

Reply:
What is done for a stored procedure that allows to create empty partitions? I think we need to follow that pattern to be consistent.

Reply:
I added an assert for empty bucketed table when path is empty (default value during initialization).  `checkState(!createFilesForMissingBuckets(table, session), "Empty location is only allowed for empty temporary table when zero-row file is not created");`